assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
Weird. Assignability was only added as a theoretical feature so we could potentially walk it back and validate equality only.